client/java/src/main/java/io/openlineage/client/metrics/MetricsResolver.java

```diff
@@ -0,0 +1,114 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.client.metrics;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.composite.CompositeMeterRegistry;
+import io.openlineage.client.utils.ReflectionUtils;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * MetricsResolver is a class that manages global OpenLineage meter registry implementation that
+ * allows integrations to both add metrics backend from common OpenLineage config, or grab an
+ * instance of an initialized MeterRegistry.
+ */
+public class MetricsResolver {
+
+  private static final List<MetricsBuilder> builders;
+  private static CompositeMeterRegistry registry;
+
+  static {
+    ServiceLoader<MetricsBuilder> loader = ServiceLoader.load(MetricsBuilder.class);
+    builders =
+        Stream.concat(
+                getMetricsBuilders().stream(), StreamSupport.stream(loader.spliterator(), false))
+            .collect(Collectors.toList());
+    registry = new CompositeMeterRegistry();
+  }
+
+  /**
+   * Parses the configuration map to create an optional MeterRegistry.
+   *
+   * @param config The configuration map for the MeterRegistry.
+   * @return Optional MeterRegistry created based on the configuration. Empty optional if the type
+   *     is invalid or empty.
+   */
+  public static Optional<MeterRegistry> parseMeterRegistryConfig(Map<String, Object> config) {
+    if (config == null) {
+      return Optional.empty();
+    }
+    Object type = config.get("type");
+    if (!(type instanceof String) || "".equals(type)) {
+      return Optional.empty();
+    }
+    return getConfigBuilder((String) type).map(x -> x.registry(config));
+  }
+
+  /**
+   * Adds a MeterRegistry to the common OpenLineage meter registry based on the provided
+   * configuration.
+   *
+   * @param config The configuration for the MeterRegistry.
+   * @return Common registry configured with the added MeterRegistry.
+   */
+  public static MeterRegistry addMeterRegistryFromConfig(Map<String, Object> config) {
```

Comment:
Would love to find later on in the docs short definition of: meter, metric, meter registry and metric builder. 
I found that non-straightforward: metric resolver class reading meter registry from config. 
If a class resolves MeterRegistry from config, shouldn't it be `MeterRegistryResolver` instead of `MetricResolver`?

Reply:
Comment is obsoleted, it's now `MeterRegistryFactory`.